examples/medplum-demo-bots/src/epic/epic-query-patient.ts

```diff
@@ -0,0 +1,68 @@
+import { BotEvent, MedplumClient } from '@medplum/core';
+import { Patient } from '@medplum/fhirtypes';
+import { SignJWT } from 'jose';
+import { randomBytes, createPrivateKey } from 'crypto';
+import fetch from 'node-fetch';
+
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<Patient> {
+  const privateKeyString = event.secrets['EPIC_PRIVATE_KEY'] as string;
+  const clientId = event.secrets['EPIC_CLIENT_ID'] as string;
+  if (!privateKeyString || !clientId) {
+    return { resourceType: 'Patient' };
+  }
+  const privateKey = createPrivateKey(privateKeyString);
+  const baseUrl = 'https://fhir.epic.com/interconnect-fhir-oauth/';
+  const tokenUrl = 'https://fhir.epic.com/interconnect-fhir-oauth/oauth2/token';
+  const fhirUrlPath = 'api/FHIR/R4/';
+
+  // Construct Epic MedplumClient base
+  const epicClient = new MedplumClient({
+    fetch,
+    baseUrl: baseUrl,
+    tokenUrl: tokenUrl,
+    fhirUrlPath: fhirUrlPath,
+    clientId: clientId,
+    onUnauthenticated: () => console.error('Unauthenticated'),
+  });
+
+  const jwt = await new SignJWT({})
+    .setProtectedHeader({ alg: 'RS384', typ: 'JWT' })
+    .setIssuer(clientId)
+    .setSubject(clientId)
+    .setAudience(baseUrl + 'oauth2/token')
+    .setJti(randomBytes(16).toString('hex'))
+    .setIssuedAt()
+    .setExpirationTime('5m')
+    .sign(privateKey);
+
+  const formBody = new URLSearchParams();
```

Comment:
Not implemented today, but for future reference...

Most cases where we hit a token endpoint, we use `MedplumClient.fetchTokens`

For most grant types, we have helper methods that setup the `formBody`.

For example, for normal client credentials flow, you can use `MedplumClient.startClientLogin`

And for one form of JWT bearer, you can use `MedplumClient.startJwtBearerLogin`

Ideally we would add `MedplumClient.startJwtAssertionLogin` which would set this up for you, call `fetchTokens`, and set the the access token

So, all put together, lines 38-52 should be a single command:

```ts
await epicClient.startJwtAssertionLogin(jwt);
```

Reply:
Filed here: https://github.com/medplum/medplum/issues/2850